File: Chat-Backend/src/main/java/com/github/sourcecase/chat/service/api/ChatDTOFactory.java
package com.github.sourcecase.chat.service.api;

import com.github.sourcecase.chat.service.api.discussion.ChatDiscussionDTO;
import com.github.sourcecase.chat.service.api.discussion.ChatMessageDTO;
import com.github.sourcecase.chat.service.api.groups.ChatGroupDTO;
import com.github.sourcecase.chat.service.api.users.ChatParticipantDTO;
import com.github.sourcecase.chat.service.api.users.ChatUserLoginDTO;

import java.sql.Time;
import java.util.List;

public interface ChatDTOFactory {

	<T> T createFromJson(Class<T> dtoType, String json);

	ChatUserLoginDTO createChatUserLoginDTO(String name, String password);

	ChatGroupDTO createChatGroupDTO(long id, String name);

	ChatParticipantDTO createChatParticipantDTO(long id, String name);

	ChatMessageDTO createChatMessageDTO(long id, String text, Time time, ChatParticipantDTO sender);

	ChatDiscussionDTO createChatDiscussionDTO(ChatGroupDTO chatGroupDTO, List<ChatMessageDTO> chatMessageDTOs);

}

File: Chat-Backend/src/main/webapp/WEB-INF/data/javascript/chatDtoFactory.js
function createChatUserLoginDto(name, password) {
	return {"type":"userLogin", "name": name, "password": password};
}

function createChatUserLoginDtoJson(username, password) {
	var userChatLoginDto = createChatUserLoginDto(username, password);
    return JSON.stringify(userChatLoginDto);
}

function createChatParticipantDto(id, name) {
	return {"type":"participant", "id": id, "name": name};
}

function createChatParticipantDtoJson(username, password) {
	var chatParticipantDto = createChatUserLoginDto(username, password);
    return JSON.stringify(chatParticipantDto);
}

function createChatMessageDto(text, chatParticipantDto) {
	return {"type":"message","id":-1, "text": text, "time":"19:09:52", "sender": chatParticipantDto};
}

function createChatMessageDtoJson(text, chatParticipantDto) {
	var chatMessageDto = createChatMessageDto(text, chatParticipantDto);
    return JSON.stringify(chatMessageDto);
}